diff -Naur pytorch-1.6.0.orig/aten/CMakeLists.txt pytorch-1.6.0/aten/CMakeLists.txt
--- pytorch-1.6.0.orig/aten/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/aten/CMakeLists.txt	2020-08-04 00:02:30.597970070 -0400
@@ -36,9 +36,9 @@
 set(ATen_HIP_DEPENDENCY_LIBS)
 set(ATen_PUBLIC_CUDA_DEPENDENCY_LIBS)
 set(ATen_PUBLIC_HIP_DEPENDENCY_LIBS)
-set(ATEN_INSTALL_BIN_SUBDIR "bin" CACHE PATH "ATen install binary subdirectory")
-set(ATEN_INSTALL_LIB_SUBDIR "lib" CACHE PATH "ATen install library subdirectory")
-set(ATEN_INSTALL_INCLUDE_SUBDIR "include" CACHE PATH "ATen install include subdirectory")
+set(ATEN_INSTALL_BIN_SUBDIR ${CMAKE_INSTALL_BINDIR} CACHE PATH "ATen install binary subdirectory")
+set(ATEN_INSTALL_LIB_SUBDIR ${CMAKE_INSTALL_LIBDIR} CACHE PATH "ATen install library subdirectory")
+set(ATEN_INSTALL_INCLUDE_SUBDIR ${CMAKE_INSTALL_INCLUDEDIR} CACHE PATH "ATen install include subdirectory")
 
 if(USE_CUDA)
   list(APPEND ATen_CUDA_INCLUDE ${CUDA_INCLUDE_DIRS})
diff -Naur pytorch-1.6.0.orig/aten/src/ATen/CMakeLists.txt pytorch-1.6.0/aten/src/ATen/CMakeLists.txt
--- pytorch-1.6.0.orig/aten/src/ATen/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/aten/src/ATen/CMakeLists.txt	2020-08-04 00:03:44.141645208 -0400
@@ -10,10 +10,10 @@
 
 # Can be compiled standalone
 if(NOT AT_INSTALL_BIN_DIR OR NOT AT_INSTALL_LIB_DIR OR NOT AT_INSTALL_INCLUDE_DIR OR NOT AT_INSTALL_SHARE_DIR)
-  set(AT_INSTALL_BIN_DIR "bin" CACHE PATH "AT install binary subdirectory")
-  set(AT_INSTALL_LIB_DIR "lib" CACHE PATH "AT install library subdirectory")
-  set(AT_INSTALL_INCLUDE_DIR "include" CACHE PATH "AT install include subdirectory")
-  set(AT_INSTALL_SHARE_DIR "share" CACHE PATH "AT install include subdirectory")
+  set(AT_INSTALL_BIN_DIR ${CMAKE_INSTALL_BINDIR} CACHE PATH "AT install binary subdirectory")
+  set(AT_INSTALL_LIB_DIR ${CMAKE_INSTALL_LIBDIR} CACHE PATH "AT install library subdirectory")
+  set(AT_INSTALL_INCLUDE_DIR ${CMAKE_INSTALL_INCLUDEDIR} CACHE PATH "AT install include subdirectory")
+  set(AT_INSTALL_SHARE_DIR ${CMAKE_INSTALL_DATAROOTDIR} CACHE PATH "AT install include subdirectory")
 endif()
 
 # this flag is used in Config but set externally, we must normalize it
diff -Naur pytorch-1.6.0.orig/aten/src/ATen/cpu/tbb/CMakeLists.txt pytorch-1.6.0/aten/src/ATen/cpu/tbb/CMakeLists.txt
--- pytorch-1.6.0.orig/aten/src/ATen/cpu/tbb/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/aten/src/ATen/cpu/tbb/CMakeLists.txt	2020-08-04 00:05:37.130131053 -0400
@@ -7,6 +7,8 @@
 include(CheckCXXCompilerFlag)
 include(CheckCXXSourceRuns)
 
+include(GNUInstallDirs)
+
 if(POLICY CMP0058)
   cmake_policy(SET CMP0058 NEW)
 endif()
@@ -25,16 +27,16 @@
   set(TBB_INSTALL_EXPORT_NAME "Caffe2Targets")
 endif()
 if(NOT TBB_INSTALL_EXPORT_DESTINATION)
-  set(TBB_INSTALL_EXPORT_DESTINATION lib)
+  set(TBB_INSTALL_EXPORT_DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 if(NOT TBB_INSTALL_RUNTIME_DIR)
-  set(TBB_INSTALL_RUNTIME_DIR bin)
+  set(TBB_INSTALL_RUNTIME_DIR ${CMAKE_INSTALL_BINDIR})
 endif()
 if(NOT TBB_INSTALL_LIBRARY_DIR)
-  set(TBB_INSTALL_LIBRARY_DIR lib)
+  set(TBB_INSTALL_LIBRARY_DIR ${CMAKE_INSTALL_LIBDIR})
 endif()
 if(NOT TBB_INSTALL_ARCHIVE_DIR)
-  set(TBB_INSTALL_ARCHIVE_DIR lib)
+  set(TBB_INSTALL_ARCHIVE_DIR ${CMAKE_INSTALL_LIBDIR})
 endif()
 if(NOT TBB_INSTALL_INCLUDE_DIR)
   set(TBB_INSTALL_INCLUDE_DIR "${TBB_ROOT_DIR}/include")
diff -Naur pytorch-1.6.0.orig/c10/cuda/CMakeLists.txt pytorch-1.6.0/c10/cuda/CMakeLists.txt
--- pytorch-1.6.0.orig/c10/cuda/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/c10/cuda/CMakeLists.txt	2020-08-04 00:07:16.273076228 -0400
@@ -61,13 +61,13 @@
 # Note: for now, we will put all export path into one single Caffe2Targets group
 # to deal with the cmake deployment need. Inside the Caffe2Targets set, the
 # individual libraries like libc10.so and libcaffe2.so are still self-contained.
-install(TARGETS c10_cuda EXPORT Caffe2Targets DESTINATION lib)
+install(TARGETS c10_cuda EXPORT Caffe2Targets DESTINATION ${CMAKE_INSTALL_LIBDIR})
 foreach(file ${C10_CUDA_HEADERS})
   get_filename_component( dir ${file} DIRECTORY )
-  install( FILES ${file} DESTINATION include/c10/cuda/${dir} )
+  install( FILES ${file} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/c10/cuda/${dir} )
 endforeach()
 install(FILES ${CMAKE_BINARY_DIR}/c10/cuda/impl/cuda_cmake_macros.h
-  DESTINATION include/c10/cuda/impl)
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/c10/cuda/impl)
 
 if(MSVC AND C10_CUDA_BUILD_SHARED_LIBS)
   install(FILES $<TARGET_PDB_FILE:c10_cuda> DESTINATION lib OPTIONAL)
diff -Naur pytorch-1.6.0.orig/c10/hip/CMakeLists.txt pytorch-1.6.0/c10/hip/CMakeLists.txt
--- pytorch-1.6.0.orig/c10/hip/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/c10/hip/CMakeLists.txt	2020-08-04 00:08:48.203216430 -0400
@@ -55,9 +55,9 @@
 add_subdirectory(test)
 
 # ---[ Installation
-install(TARGETS c10_hip EXPORT Caffe2Targets DESTINATION lib)
+install(TARGETS c10_hip EXPORT Caffe2Targets DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
-        DESTINATION include
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
         FILES_MATCHING PATTERN "*.h")
 install(FILES ${CMAKE_BINARY_DIR}/c10/hip/impl/hip_cmake_macros.h
-  DESTINATION include/c10/hip/impl)
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/c10/hip/impl)
diff -Naur pytorch-1.6.0.orig/caffe2/CMakeLists.txt pytorch-1.6.0/caffe2/CMakeLists.txt
--- pytorch-1.6.0.orig/caffe2/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/caffe2/CMakeLists.txt	2020-08-04 00:13:37.057025651 -0400
@@ -260,15 +260,15 @@
 set(TORCH_ROOT "${TORCH_SRC_DIR}/..")
 
 if(NOT TORCH_INSTALL_BIN_DIR)
-  set(TORCH_INSTALL_BIN_DIR bin)
+  set(TORCH_INSTALL_BIN_DIR ${CMAKE_INSTALL_BINDIR})
 endif()
 
 if(NOT TORCH_INSTALL_INCLUDE_DIR)
-  set(TORCH_INSTALL_INCLUDE_DIR include)
+  set(TORCH_INSTALL_INCLUDE_DIR ${CMAKE_INSTALL_INCLUDEDIR})
 endif()
 
 if(NOT TORCH_INSTALL_LIB_DIR)
-  set(TORCH_INSTALL_LIB_DIR lib)
+  set(TORCH_INSTALL_LIB_DIR ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 
@@ -1249,7 +1249,7 @@
   if(${BUILDING_WITH_TORCH_LIBS})
     # site-packages/caffe2/python/caffe2_pybind11_state
     # site-packages/torch/lib
-    set(caffe2_pybind11_rpath "${_rpath_portable_origin}/../../torch/lib")
+    set(caffe2_pybind11_rpath "${_rpath_portable_origin}/../../torch/${CMAKE_INSTALL_LIBDIR}")
   endif(${BUILDING_WITH_TORCH_LIBS})
 
   # Must also include `CMAKE_SHARED_LINKER_FLAGS` in linker flags for
diff -Naur pytorch-1.6.0.orig/caffe2/proto/CMakeLists.txt pytorch-1.6.0/caffe2/proto/CMakeLists.txt
--- pytorch-1.6.0.orig/caffe2/proto/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/caffe2/proto/CMakeLists.txt	2020-08-04 00:14:05.315133367 -0400
@@ -16,4 +16,4 @@
 target_compile_definitions(
     Caffe2_PROTO PRIVATE ${Caffe2_API_DEFINE})
 
-install(FILES ${Caffe2_PROTO_HEADERS} DESTINATION include/caffe2/proto)
+install(FILES ${Caffe2_PROTO_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/caffe2/proto)
diff -Naur pytorch-1.6.0.orig/CMakeLists.txt pytorch-1.6.0/CMakeLists.txt
--- pytorch-1.6.0.orig/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/CMakeLists.txt	2020-08-04 00:01:19.618611037 -0400
@@ -112,6 +112,9 @@
   endif()
 endif()
 
+# Make installation paths FHS-compliant as much, as possible
+include(GNUInstallDirs)
+
 # ---[ Options.
 # Note to developers: if you add an option below, make sure you also add it to
 # cmake/Summary.cmake so that the summary prints out the option values.
diff -Naur pytorch-1.6.0.orig/modules/detectron/CMakeLists.txt pytorch-1.6.0/modules/detectron/CMakeLists.txt
--- pytorch-1.6.0.orig/modules/detectron/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/modules/detectron/CMakeLists.txt	2020-08-04 00:15:25.790979903 -0400
@@ -16,7 +16,7 @@
         ${Detectron_GPU_SRCS})
 
     target_link_libraries(caffe2_detectron_ops_gpu torch ${OpenMP_link})
-    install(TARGETS caffe2_detectron_ops_gpu DESTINATION lib)
+    install(TARGETS caffe2_detectron_ops_gpu DESTINATION ${CMAKE_INSTALL_LIBDIR})
     if(MSVC)
       install(FILES $<TARGET_PDB_FILE:caffe2_detectron_ops_gpu> DESTINATION lib OPTIONAL)
     endif()
@@ -29,7 +29,7 @@
         ${Detectron_HIP_SRCS})
     target_compile_options(caffe2_detectron_ops_hip PRIVATE ${HIP_CXX_FLAGS})
     target_link_libraries(caffe2_detectron_ops_hip torch)
-    install(TARGETS caffe2_detectron_ops_hip DESTINATION lib)
+    install(TARGETS caffe2_detectron_ops_hip DESTINATION ${CMAKE_INSTALL_LIBDIR})
   elseif(NOT IOS_PLATFORM)
     add_library(caffe2_detectron_ops SHARED ${Detectron_CPU_SRCS})
     if(HAVE_SOVERSION)
@@ -37,7 +37,7 @@
         VERSION ${TORCH_VERSION} SOVERSION ${TORCH_SOVERSION})
     endif()
     target_link_libraries(caffe2_detectron_ops torch ${OpenMP_link})
-    install(TARGETS caffe2_detectron_ops DESTINATION lib)
+    install(TARGETS caffe2_detectron_ops DESTINATION ${CMAKE_INSTALL_LIBDIR})
     if(MSVC)
       install(FILES $<TARGET_PDB_FILE:caffe2_detectron_ops> DESTINATION lib OPTIONAL)
     endif()
diff -Naur pytorch-1.6.0.orig/modules/module_test/CMakeLists.txt pytorch-1.6.0/modules/module_test/CMakeLists.txt
--- pytorch-1.6.0.orig/modules/module_test/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/modules/module_test/CMakeLists.txt	2020-08-04 00:16:04.824268692 -0400
@@ -16,7 +16,7 @@
         VERSION ${TORCH_VERSION} SOVERSION ${TORCH_SOVERSION})
   endif()
   target_link_libraries(caffe2_module_test_dynamic torch_library)
-  install(TARGETS caffe2_module_test_dynamic DESTINATION lib)
+  install(TARGETS caffe2_module_test_dynamic DESTINATION ${CMAKE_INSTALL_LIBDIR})
   if(MSVC AND BUILD_SHARED_LIBS)
     install(FILES $<TARGET_PDB_FILE:caffe2_module_test_dynamic> DESTINATION lib OPTIONAL)
   endif()
diff -Naur pytorch-1.6.0.orig/modules/observers/CMakeLists.txt pytorch-1.6.0/modules/observers/CMakeLists.txt
--- pytorch-1.6.0.orig/modules/observers/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/modules/observers/CMakeLists.txt	2020-08-04 00:16:38.958018053 -0400
@@ -21,7 +21,7 @@
 target_link_libraries(caffe2_observers PUBLIC torch_library)
 target_include_directories(caffe2_observers PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/..)
 target_compile_options(caffe2_observers PRIVATE "-DCAFFE2_BUILD_OBSERVER_LIB")
-install(TARGETS caffe2_observers DESTINATION lib)
+install(TARGETS caffe2_observers DESTINATION ${CMAKE_INSTALL_LIBDIR})
 caffe2_interface_library(caffe2_observers caffe2_observers_library)
 if(MSVC AND BUILD_SHARED_LIBS)
   install(FILES $<TARGET_PDB_FILE:caffe2_observers> DESTINATION lib OPTIONAL)
diff -Naur pytorch-1.6.0.orig/modules/rocksdb/CMakeLists.txt pytorch-1.6.0/modules/rocksdb/CMakeLists.txt
--- pytorch-1.6.0.orig/modules/rocksdb/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/modules/rocksdb/CMakeLists.txt	2020-08-04 00:17:21.354673646 -0400
@@ -59,7 +59,7 @@
 target_link_libraries(caffe2_rocksdb PUBLIC torch_library)
 target_link_libraries(caffe2_rocksdb PRIVATE ${RocksDB_LIBRARIES})
 target_include_directories(caffe2_rocksdb PRIVATE ${RocksDB_INCLUDE_DIR})
-install(TARGETS caffe2_rocksdb DESTINATION lib)
+install(TARGETS caffe2_rocksdb DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 # ---[ Last, Append the library to Caffe2_MODULES, if we are building with
 # the main repo.
diff -Naur pytorch-1.6.0.orig/test/cpp/api/CMakeLists.txt pytorch-1.6.0/test/cpp/api/CMakeLists.txt
--- pytorch-1.6.0.orig/test/cpp/api/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/test/cpp/api/CMakeLists.txt	2020-08-04 00:18:24.192571088 -0400
@@ -67,7 +67,7 @@
 endif()
 
 if(INSTALL_TEST)
-  install(TARGETS test_api DESTINATION bin)
+  install(TARGETS test_api DESTINATION ${CMAKE_INSTALL_BINDIR})
   # Install PDB files for MSVC builds
   if(MSVC AND BUILD_SHARED_LIBS)
     install(FILES $<TARGET_PDB_FILE:test_api> DESTINATION bin OPTIONAL)
diff -Naur pytorch-1.6.0.orig/test/cpp/dist_autograd/CMakeLists.txt pytorch-1.6.0/test/cpp/dist_autograd/CMakeLists.txt
--- pytorch-1.6.0.orig/test/cpp/dist_autograd/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/test/cpp/dist_autograd/CMakeLists.txt	2020-08-04 00:19:33.534178557 -0400
@@ -20,7 +20,7 @@
   endif()
 
   if(INSTALL_TEST)
-    install(TARGETS test_dist_autograd DESTINATION bin)
+    install(TARGETS test_dist_autograd DESTINATION ${CMAKE_INSTALL_BINDIR})
     # Install PDB files for MSVC builds
     if(MSVC AND BUILD_SHARED_LIBS)
       install(FILES $<TARGET_PDB_FILE:test_dist_autograd> DESTINATION bin OPTIONAL)
diff -Naur pytorch-1.6.0.orig/test/cpp/jit/CMakeLists.txt pytorch-1.6.0/test/cpp/jit/CMakeLists.txt
--- pytorch-1.6.0.orig/test/cpp/jit/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/test/cpp/jit/CMakeLists.txt	2020-08-04 00:20:32.571652434 -0400
@@ -80,7 +80,7 @@
 endif()
 
 if(INSTALL_TEST)
-  install(TARGETS test_jit DESTINATION bin)
+  install(TARGETS test_jit DESTINATION ${CMAKE_INSTALL_BINDIR})
   # Install PDB files for MSVC builds
   if(MSVC AND BUILD_SHARED_LIBS)
     install(FILES $<TARGET_PDB_FILE:test_jit> DESTINATION bin OPTIONAL)
diff -Naur pytorch-1.6.0.orig/torch/CMakeLists.txt pytorch-1.6.0/torch/CMakeLists.txt
--- pytorch-1.6.0.orig/torch/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/torch/CMakeLists.txt	2020-08-04 00:21:59.199146626 -0400
@@ -21,7 +21,7 @@
 set(TORCH_ROOT "${TORCH_SRC_DIR}/..")
 
 if(NOT TORCH_INSTALL_LIB_DIR)
-  set(TORCH_INSTALL_LIB_DIR lib)
+  set(TORCH_INSTALL_LIB_DIR ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 if(MSVC)
diff -Naur pytorch-1.6.0.orig/torch/lib/c10d/CMakeLists.txt pytorch-1.6.0/torch/lib/c10d/CMakeLists.txt
--- pytorch-1.6.0.orig/torch/lib/c10d/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/torch/lib/c10d/CMakeLists.txt	2020-08-04 00:23:06.755546512 -0400
@@ -142,7 +142,7 @@
 
 target_link_libraries(c10d PUBLIC ${C10D_LIBS})
 
-install(TARGETS c10d DESTINATION lib)
+install(TARGETS c10d DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 option(BUILD_EXAMPLES "Build examples" OFF)
 if(BUILD_EXAMPLES)
@@ -156,4 +156,4 @@
 endif()
 
 # Install all header files that were prepared in the build directory
-install(DIRECTORY ${CMAKE_BINARY_DIR}/include/ DESTINATION include)
+install(DIRECTORY ${CMAKE_BINARY_DIR}/include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
diff -Naur pytorch-1.6.0.orig/torch/lib/libshm/CMakeLists.txt pytorch-1.6.0/torch/lib/libshm/CMakeLists.txt
--- pytorch-1.6.0.orig/torch/lib/libshm/CMakeLists.txt	2020-07-24 18:18:09.000000000 -0400
+++ pytorch-1.6.0/torch/lib/libshm/CMakeLists.txt	2020-08-04 00:24:57.466665693 -0400
@@ -2,11 +2,13 @@
 cmake_minimum_required(VERSION 2.6 FATAL_ERROR)
 cmake_policy(VERSION 2.6)
 
+include(GNUInstallDirs)
+
 set(TORCH_ROOT ${CMAKE_CURRENT_LIST_DIR}/../../../)
 include(${TORCH_ROOT}/cmake/public/threads.cmake)
 
 if(NOT LIBSHM_INSTALL_LIB_SUBDIR)
-  set(LIBSHM_INSTALL_LIB_SUBDIR "lib" CACHE PATH "libshm install library directory")
+  set(LIBSHM_INSTALL_LIB_SUBDIR ${CMAKE_INSTALL_LIBDIR} CACHE PATH "libshm install library directory")
 endif()
 
 # Flags
@@ -36,7 +38,7 @@
 
 add_executable(torch_shm_manager manager.cpp)
 target_link_libraries(torch_shm_manager shm)
-set_target_properties(torch_shm_manager PROPERTIES INSTALL_RPATH "${_rpath_portable_origin}/../lib")
+set_target_properties(torch_shm_manager PROPERTIES INSTALL_RPATH "${_rpath_portable_origin}/../${CMAKE_INSTALL_LIBDIR}")
 ### Torch packages supposes libraries prefix is "lib"
 set_target_properties(shm PROPERTIES
   PREFIX "lib"
@@ -81,5 +83,5 @@
 
 
 install(TARGETS shm LIBRARY DESTINATION ${LIBSHM_INSTALL_LIB_SUBDIR})
-install(FILES libshm.h DESTINATION "include")
-install(TARGETS torch_shm_manager DESTINATION "bin")
+install(FILES libshm.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(TARGETS torch_shm_manager DESTINATION ${CMAKE_INSTALL_BINDIR})
